### PR TITLE
Add newer edition of Learn Azure in a Month of Lunches

### PR DESCRIPTION
**Pull request by ItsGageH**
## What does this PR do?
Add resource(s)

### Description
Add newer edition of book

### Why is this valuable (or not)?
Newer, up-to-date information

### How do we know it's really free?
Hosted and provided by Azure

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book, that is also published in paper form.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
one link is enough

**Comment by ItsGageH** [comment]
Hi @eshellman, can you confirm what changes need doing? Remove the old (1st edition) link and only keep the newer?

**Comment by eshellman** [comment]
Remove the old (1st edition) link and only keep the newer.

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -172,7 +172,8 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Distributed Systems 3rd edition](https://www.distributed-systems.net/index.php/books/ds3/) - Maarten van Steen, Andrew S. Tanenbaum *(email address required)*
 * [How to optimize storage costs using Amazon S3](https://aws.amazon.com/s3/cloud-storage-cost-optimization-ebook/) - AWS (PDF)
 * [Kubernetes Hardening Guidance](https://media.defense.gov/2021/Aug/03/2002820425/-1/-1/1/CTR_KUBERNETESHARDENINGGUIDANCE.PDF) - NSA, CISA (PDF)
-* [Learn Azure in a Month of Lunches](https://azure.microsoft.com/mediahandler/files/resourcefiles/learn-azure-in-a-month-of-lunches/Learn_Azure_in_a_Month_of_Lunches.pdf) - Iain Foulds (PDF)
+* [Learn Azure in a Month of Lunches (2020)](https://clouddamcdnprodep.azureedge.net/gdc/2014519/original) - Iain Foulds (PDF) (2nd Edition)
+    * [Learn Azure in a Month of Lunches (2018)](https://azure.microsoft.com/mediahandler/files/resourcefiles/learn-azure-in-a-month-of-lunches/Learn_Azure_in_a_Month_of_Lunches.pdf) - Iain Foulds (PDF) (1st Edition)
```
Comment (shrn01) [inline books/free-programming-books-subjects.md:176]:
```suggestion
* [Learn Azure in a Month of Lunches](https://clouddamcdnprodep.azureedge.net/gdc/2014519/original) - Iain Foulds (PDF)
```

**Pull request merged**